fakesmtp/fakesmtp.go

```diff
@@ -0,0 +1,98 @@
+package fakesmtp
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"net"
+	"strings"
+	"time"
+)
+
+var LastMessage string
+
+type Client struct {
+	conn    net.Conn
+	address string
+	time    int64
+	bufin   *bufio.Reader
+	bufout  *bufio.Writer
+}
+
+func (c *Client) write(s string) {
+	c.bufout.WriteString(s + "\r\n")
+	c.bufout.Flush()
+}
+func (c *Client) read() string {
+	reply, err := c.bufin.ReadString('\n')
+
+	if err != nil {
+		fmt.Println("e ", err)
+	}
+	return reply
+}
+
+func addToMessage(text string) {
+	LastMessage = LastMessage + text
+}
+
+func handleClient(c *Client) {
+	LastMessage = ""
+	c.write("220 Welcome")
+	text := c.read()
+	addToMessage(text)
+	c.write("250 Received")
+	text = c.read()
+	addToMessage(text)
+	c.write("250 Sender")
+
+	text = c.read()
+	addToMessage(text)
+
+	c.write("250 Recipient")
+	text = c.read()
+	for strings.Contains(text, "RCPT") {
+		addToMessage(text)
+		c.write("250 Recipient")
+		text = c.read()
+	}
+
+	c.write("354 Ok Send data ending with <CRLF>.<CRLF>")
+
+	for {
+		text = c.read()
+		bytes := []byte(text)
+		addToMessage(text)
+		// 46 13 10
+		if bytes[0] == 46 && bytes[1] == 13 && bytes[2] == 10 {
+			break
+		}
+	}
+	c.write("250 server has transmitted the message")
+	c.conn.Close()
+}
+
+//Start listens for connections on the given port
+func WaitForMessage(port string) {
```

Comment:
This doesn't actually "wait". :) You may want to not use a goroutine here, but instead let it block. That way, people can decide whether to let it block or not. 

I would also probably change it back to `Start` and have it return an error.